### PR TITLE
🛠️ - Typescript rules against: enums & interfaces

### PR DESCRIPTION
**Pull request by MaartenBruggink**
Added two Typescript specific eslint rules;
- `no-restricted-syntax - TSEnumDeclaration` - Error when using Enums, let's push for `as const` since it's more declarative that it outputs JS.
- `@typescript-eslint/consistent-type-definitions` - Error when using interfaces. The difference between interface and type is minimal and [the one additional feature it supports](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) we shouldn't use.

Set nvm version to `v18`

### CODE REVIEW

File: eslint/configs/typescript.js [hunk 1]
```diff
@@ -4,4 +4,15 @@ module.exports = {
 
   // Full list: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
   extends: ["plugin:@typescript-eslint/recommended"],
+
+  rules: {
+    "no-restricted-syntax": [
+      "error",
+      {
+        selector: "TSEnumDeclaration",
+        message: "Don't declare enums",
```
Comment (PauloMFJ) [inline eslint/configs/typescript.js:13]: Can we change message to something for descriptive of our recommendation?

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

**Review by PauloMFJ — COMMENTED** [review]
These two rules partially go against the grain on what this package was trying to be:

```
Extended ESLint Rules
The idea behind this config is to enforce consistency across all projects. We've tried to keep the rules as minimal as possible, and for the most part simply inherit from the recommended rules of the plugins we use. The main exceptions are when it comes to the following. Note: We've purposely only defined rules that are auto fixable, these rules should make it easier to write code, and not get in the way.
```

Can we update the README to reflect the changes too? Would be nice to also explain the reasoning there.

File: eslint/configs/typescript.js (hunk 1 above)
Comment (MaartenBruggink) [reply eslint/configs/typescript.js]: Changed it a bit, not sure about it yet tho, let me know what you think

**Review by PauloMFJ — APPROVED** [review]

File: eslint/configs/typescript.js (hunk 1 above)
Comment (PauloMFJ) [reply eslint/configs/typescript.js]: I think it's fine. Would maybe replace `X` with `Foo` though? Just a bit easier read I think:
```
Use `const Foo as const` with `typeof Foo` over Enum"
```

File: README.md [hunk 2]
```diff
@@ -108,6 +108,16 @@ This is used to auto fix some common inconsistencies. The following rules have b
 - [import/newline-after-import](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/newline-after-import.md) - Makes sure there’s a newline after the imports.
 - [import/no-duplicates](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md) - Merges import statements from the same file.
 
+#### [@typescript-eslint/consistent-type-definitions](https://www.npmjs.com/package/@typescript-eslint/eslint-plugin)
+
+Used to enforce `type` over `interface`, the difference between interface and type is minimal and the one additional feature it supports we shouldn't use.
```
[the remark this answers is not in the document]
Comment (MaartenBruggink) [reply README.md]: Nice one, thought I did that 👍

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@buildinams/lint",
   "description": "BiA's base lint config(s).",
-  "version": "0.2.1",
```
Comment (PauloMFJ) [inline package.json:4]: Can you install this? Need a package-lock change for publish

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

**Review by PauloMFJ — APPROVED** [review]

**Pull request merged**